src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -86,5 +113,260 @@ private protected IReadOnlyList<PackageReference> GetTransitivePackageReferences
+            foreach (LockFileTargetLibrary lib in graph.Libraries)
+            {
+                if (lib.Type == LibraryType.Package.Value
+                    && string.Equals(lib.Name, current.PackageIdentity.Id, StringComparison.OrdinalIgnoreCase)
+                    && ((current.HasAllowedVersions && current.AllowedVersions.Satisfies(lib.Version)) ||
+                        (current.PackageIdentity.HasVersion && current.PackageIdentity.Version.Equals(lib.Version))))
+                {
+                    node = lib;
+                    break;
+                }
+            }
+
+            if (node != default)
+            {
+                visited.Add(current.PackageIdentity); // visited
+
+                // Update cache
+                TransitiveEntry cachedEntry = GetCachedTransitiveOrigin(current.PackageIdentity);
+
+                if (cachedEntry == null)
+                {
+                    cachedEntry = new Dictionary<FrameworkRuntimePair, IList<PackageReference>>
+                    {
+                        [fxRidEntry] = new List<PackageReference>()
+                    };
+                }
+
+                if (!cachedEntry.ContainsKey(fxRidEntry))
+                {
+                    cachedEntry[fxRidEntry] = new List<PackageReference>();
+                }
+
+                if (!cachedEntry[fxRidEntry].Contains(top))
+                {
+                    cachedEntry[fxRidEntry].Add(top);
+                }
+
+                SetCachedTransitiveOrigin(current.PackageIdentity, cachedEntry);
+
+                foreach (PackageDependency dep in node.Dependencies.ToList()) // Casting to list to prevent backing allocations
+                {
+                    // Create PackageReference object as a data-model based on dependency
+                    var pkgChild = new PackageReference(
+                        identity: new PackageIdentity(dep.Id, dep.VersionRange.MinVersion),
+                        targetFramework: fxRidEntry.Framework,
+                        userInstalled: false,
+                        developmentDependency: false,
+                        requireReinstallation: false,
+                        allowedVersions: dep.VersionRange);
+
+                    if (!visited.Contains(pkgChild.PackageIdentity))
+                    {
+                        MarkTransitiveOrigin(top, pkgChild, graph, visited, fxRidEntry, token);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Generates a cache key for Transitive Originas cache
+        /// </summary>
+        /// <param name="transitivePackage"></param>
+        /// <returns>A string with given key</returns>
+        /// <seealso cref="GetCachedTransitiveOrigin(PackageIdentity)"/>
+        /// <seealso cref="SetCachedTransitiveOrigin(PackageIdentity, TransitiveEntry)"/>
+        internal PackageIdentity GetTransitiveCacheKey(PackageIdentity transitivePackage)
```

Comment:
Do you still need this? 

It's returning itself so likely the method can just be removed.

Reply:
Removed.